var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -637,7 +637,7 @@ def build_test(self):
     def install(self, spec, prefix):
         self.generator("install")
 
-        if spec.satisfies("%fj"):
+        if spec.satisfies("%fj") or spec.satisfies("%clang@11:"):
```

Comment:
This extends https://github.com/spack/spack/pull/16864, where there was some concern about patching this for other platforms. Limiting it to `%fj` made it unlikely that it would affect builds on other machines, but expanding to `%clang` means it will affect builds on other machines (or rather, that `FindMPI.cmake` will be changed on other machines, which may not have any particular result).

I'm wondering if we should just universally set the `MPI_C_COMPILER` variable from all `mpi` providers, which I think would avoid the need for this; a limited version of that would be changing `setup_dependent_build/run_environment` within `fujitsu-mpi` specifically. Would you be able to try that out? I'd like to think on alternatives if that doesn't work for you.

Side note: `FindMPI.cmake` in `cmake@3.27.9` appears to have a dedicated section for Fujitsu:

```
#Fujitsu cross/own compiler names
set(_MPI_Fujitsu_C_COMPILER_NAMES        mpifccpx mpifcc)
set(_MPI_Fujitsu_CXX_COMPILER_NAMES      mpiFCCpx mpiFCC)
set(_MPI_Fujitsu_Fortran_COMPILER_NAMES  mpifrtpx mpifrt)
```

but I'm guessing that isn't useful since it only applies if the compiler is identified as Fujitsu (it implies to me though that we don't need the `if spec.satisfies(%fj)` part).